### PR TITLE
Add support for MRPC dataset with unit tests

### PR DESCRIPTION
**Pull request by VirgileHlav**
## Summary
- Added support for MRPC dataset
- Added mocked unit tests for MRPC dataset 

## Test
- ` pytest test/datasets/test_mrpc.py`

## Context
See #1710

### CODE REVIEW

**Review by parmeet — APPROVED** [review]
LGTM!

**Comment by Nayef211** [comment]
@ejguan @NivekT it seems like some of the CI builds are failing at the step where we try to install `torchdata` from source (i.e. [here](https://app.circleci.com/pipelines/github/pytorch/text/5488/workflows/97622483-c2d6-41bc-a20d-eea8f9cbc4a4/jobs/185386)). Do you know if there's been any recent changes that would cause this?

**Comment by ejguan** [comment]
> @ejguan @NivekT it seems like some of the CI builds are failing at the step where we try to install `torchdata` from source (i.e. [here](https://app.circleci.com/pipelines/github/pytorch/text/5488/workflows/97622483-c2d6-41bc-a20d-eea8f9cbc4a4/jobs/185386)). Do you know if there's been any recent changes that would cause this?

This seems weird to me. TorchData is required PyTorch to run some codegen. Based on the script (https://github.com/pytorch/text/blob/main/.circleci/unittest/linux/scripts/install.sh), it seems PyTorch is installed before TorchData. Not sure why this happens.
Please correct me if I am wrong. It seems the CI shouldn't use source to install TorchData https://github.com/pytorch/text/blob/ab76a046919e6358c7038512f3ee500dd17fb632/.circleci/unittest/linux/scripts/install.sh#L32-L33.

**Comment by NivekT** [comment]
> > @ejguan @NivekT it seems like some of the CI builds are failing at the step where we try to install `torchdata` from source (i.e. [here](https://app.circleci.com/pipelines/github/pytorch/text/5488/workflows/97622483-c2d6-41bc-a20d-eea8f9cbc4a4/jobs/185386)). Do you know if there's been any recent changes that would cause this?
> 
> This seems weird to me. TorchData is required PyTorch to run some codegen. Based on the script (https://github.com/pytorch/text/blob/main/.circleci/unittest/linux/scripts/install.sh), it seems PyTorch is installed before TorchData. Not sure why this happens. Please correct me if I am wrong. It seems the CI shouldn't use source to install TorchData
> 
> https://github.com/pytorch/text/blob/ab76a046919e6358c7038512f3ee500dd17fb632/.circleci/unittest/linux/scripts/install.sh#L32-L33
> 
Maybe switching to install nightly with conda will help?

**Comment by Nayef211** [comment]
> Please correct me if I am wrong. It seems the CI shouldn't use source to install TorchData

@ejguan my mistake I guess we are relying on torchdata nightlies in our unit tests setup and not installing directly from source.

> Maybe switching to install nightly with conda will help?

@NivekT do you know what the correct command would be to install the nightly from conda?
https://github.com/pytorch/text/blob/ab76a046919e6358c7038512f3ee500dd17fb632/.circleci/unittest/linux/scripts/install.sh#L32-L33

**Comment by Nayef211** [comment]
> @NivekT do you know what the correct command would be to install the nightly from conda?


Nvm I just found the following link https://anaconda.org/pytorch-nightly/torchdata. I assume it would be as follows
```
conda install -c pytorch-nightly torchdata
```

**Comment by ejguan** [comment]
I think this command should be fine:
`pip install --pre torchdata --extra-index-url https://download.pytorch.org/whl/nightly/cpu`
I just tested it on my local machine.

File: test/datasets/test_mrpc.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                rand_string_1 = get_random_unicode(seed)
+                rand_string_2 = get_random_unicode(seed + 1)
+                dataset_line = (label, rand_string_1, rand_string_2)
+                f.write(f"{label}\t{i}\t{i}\t{rand_string_1}\t{rand_string_2}\n")
+
+                # append line to correct dataset split
+                mocked_data[file_type].append(dataset_line)
+                seed += 1
+
+    print(mocked_data)
```
Comment (Nayef211) [inline test/datasets/test_mrpc.py:36]: nit: should we delete this?

**Review by Nayef211 — APPROVED** [review]
LGTM. Delete `print` lines before merging

**Pull request merged**